### PR TITLE
test(cypress): add integration tests

### PR DESCRIPTION
**Pull request by ismay**
https://jira.dhis2.org/browse/DHIS2-10503

#### Testing lib docs

- https://testing-library.com/docs/cypress-testing-library/intro
- https://testing-library.com/docs/queries/about#priority

#### Abstraction

There's some repetition in the test code. I've been trying to follow suggestions like the ones found in the cypress docs: https://docs.cypress.io/api/cypress-api/custom-commands.html#Best-Practices

> Testing in Cypress is all about readability and simplicity. You don’t have to do that much actual programming to get a lot done. You also don’t need to worry about keeping your code as DRY as possible. Test code serves a different purpose than app code. Understandability and debuggability should be prioritized above all else.

Let me know if you feel the code could be made more readable.

#### Todo

- [x] List route
- [x] View route
- [x] Not authorized route
- [x] Add route
- [x] Edit route

#### Follow up

- [ ] Update components that aren't accessible in ui, and replace component in `Temporary` with ui components (see TODO comments here 0596432)
- [ ] Update tests to use more accessible selectors

### CODE REVIEW

File: cypress/integration/edit-route/info-link/index.js [hunk 1]
```diff
@@ -0,0 +1,23 @@
+import { Given, Then } from 'cypress-cucumber-preprocessor/steps'
+
+const infoHref =
```
Comment (mediremi) [inline cypress/integration/edit-route/info-link/index.js:3]: I guess this is off-topic to this this PR but this link redirects to https://docs.dhis2.org/en/home.html :disappointed:

**Review by mediremi — APPROVED** [review]
LGTM :heavy_check_mark: 

I particularly like how the accessible selectors and helpers make the tests easier to read. I hope all future cypress tests across our apps are written like this :ok_hand:

File: cypress/integration/edit-route/info-link/index.js (hunk 1 above)
Comment (ismay) [reply cypress/integration/edit-route/info-link/index.js:3]: Yeah exactly. Phil has merged the docs PR here: https://github.com/dhis2/dhis2-docs/pull/631, so it should all work once 2.36 has been released.

**Comment by ismay** [comment]
> LGTM ✔️
> 
> I particularly like how the accessible selectors and helpers make the tests easier to read. I hope all future cypress tests across our apps are written like this 👌

Yeah I agree, I like it as well. Not too much of a hassle to set up either. All in all quite a useful lib.

**Pull request merged**